oc-backup.sh

```diff
@@ -31,17 +31,14 @@ done
 
 
 ### Databases ###
-for i in `oc get projects --no-headers |grep Active |awk '{print $1}'`
+for PROJECT in `oc get projects --no-headers |grep Active |awk '{print $1}'`
 do
-  oc observe -n $i --once pods \
-    -a '{ .metadata.labels.deploymentconfig }'   \
-    -a '{ .metadata.labels.backup     }' \
-    -a '{ .metadata.labels.backupvolumemount     }'   -- echo \
-   |grep -v ^# \
-   |while read PROJECT POD DC BACKUP BACKUPVOLUMEMOUNT
+  oc get pods -n $PROJECT --output=go-template='{{range .items}}{{if .metadata.labels.backup}}{{.metadata.name}} {{if .metadata.labels.deploymentconfig}}{{.metadata.labels.deploymentconfig}}{{else}}{{.metadata.labels.app}}{{end}} {{.metadata.labels.backup}} {{.metadata.labels.backupvolumemount}}{{"\n"}}{{end}}{{end}}' \
+   |while read POD DC BACKUP BACKUPVOLUMEMOUNT
   do
     [ "$BACKUP" == "" ] && continue
-    echo "$POD in $PROJECT has the following BACKUP label: $BACKUP"
+    [ "$DC" == "" ] && echo "No app or deploymentconfig label for pod $POD" && continue
```

Comment:
This line would never trigger i think, since we "read" all values in the while loop line by line.
If line 36 "oc get pods" would not output a DC the backup and backupvolumemount label would slide forward - so backup would be stored in $DC and backupvolumemount in $BACKUP.

The case you're trying to catch is when no deploymentconfig and no app-label is set, right?